asmcomp/arm64/emit.mlp

```diff
@@ -464,23 +464,46 @@ module BR = Branch_relaxation.Make (struct
     | Lop (Itailcall_imm { func; _ }) ->
       if func = f.fun_name then 1 else epilogue_size f
     | Lop (Iextcall {alloc; stack_ofs} ) ->
-      if stack_ofs > 0 then 5
-      else if alloc then 3
-      else 7
+      if !Clflags.alternate_code_generation then begin
+        if stack_ofs > 0 then 6
+        else if alloc then 4
+        else 8
+      end else begin
+        if stack_ofs > 0 then 5
+        else if alloc then 3
+        else 7
+      end
     | Lop (Istackoffset _) -> 2
-    | Lop (Iload  { memory_chunk; addressing_mode; is_atomic }) ->
-      let based = match addressing_mode with Iindexed _ -> 0 | Ibased _ -> 1
-      and barrier = if is_atomic then 1 else 0
-      and single = match memory_chunk with Single -> 2 | _ -> 1 in
-      based + barrier + single
+    | Lop (Iload  { memory_chunk; addressing_mode; is_atomic; mutability }) ->
+      if !Clflags.alternate_code_generation then begin
+        let based = match addressing_mode with Iindexed _ -> 0 | Ibased _ -> 1
+        and insns = match memory_chunk with
+          | Single -> 2
+          | (Word_int | Word_val)
+              when is_atomic || mutability = Asttypes.Mutable -> 2
+          | _ -> 1
+        in
+        based + insns
+      end else begin
+        let based = match addressing_mode with Iindexed _ -> 0 | Ibased _ -> 1
+        and barrier = if is_atomic then 1 else 0
+        and single = match memory_chunk with Single -> 2 | _ -> 1 in
+        based + barrier + single
+      end
     | Lop (Istore (memory_chunk, addressing_mode, assignment)) ->
-      let based = match addressing_mode with Iindexed _ -> 0 | Ibased _ -> 1
-      and barrier =
-        match memory_chunk, assignment with
-        | (Word_int | Word_val), true -> 1
-        | _ -> 0
-      and single = match memory_chunk with Single -> 2 | _ -> 1 in
-      based + barrier + single
+      if !Clflags.alternate_code_generation then begin
+        let based = match addressing_mode with Iindexed _ -> 0 | Ibased _ -> 1
```

Comment:
Nit: Move `based` and `single` out of the body of if and else to avoid having them duplicated twice in code.